tests/run/type_inference.pyx

```diff
@@ -531,6 +531,12 @@ def safe_only():
     cdef int c_int = 1
     assert typeof(abs(c_int)) == "int", typeof(abs(c_int))
 
+    # float can be safely inferred to double
+    cdef float fl = 5.0
+    from_fl = fl
+    assert typeof(from_fl) == "double", typeof(from_fl)
```

Comment:
This seems … surprising. Why should we infer `double` for a clear C `float`? It's one thing to reduce Python `float` to C `double`, but C `float` should not be affected by this.